sdk/src/http/restricted.rs

```diff
@@ -68,7 +76,7 @@ impl<T> RestrictedResolver<T> {
     }
 
     /// Creates a new `RestrictedResolver` with the specified allowed list.
-    #[allow(dead_code)] // TODO: temp until http module is public
+    #[allow(dead_code)] // Public API, not used internally
```

Comment:
Note: when `pub(crate)`->`pub` on the `http` module is changed in `lib.rs` we will no longer need this. I think we should make that change in this PR since HTTP resolvers will be part of the public API in the `Context`.